x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_list.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import { EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { Immutable } from '../../../../../common/endpoint/types';
+import { TrustedApp } from '../../../../../common/endpoint/types/trusted_apps';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { getTrustedAppsListPath } from '../../../common/routing';
+
+import {
+  getListCurrentPageIndex,
+  getListCurrentPageSize,
+  getListErrorMessage,
+  getListItems,
+  getListTotalItemsCount,
+  isListLoading,
+} from '../store/selectors';
+
+import { useTrustedAppsSelector } from './hooks';
+
+import { FormattedDate } from '../../../../common/components/formatted_date';
+
+const OS_TITLES: { [K in TrustedApp['os']]: string } = {
```

Comment:
Consider moving `OS_TITLES`, `COLUMN_TITLES `, `COLUMN_DEFINITIONS ` into the `TrustedAppsList` component below to avoid the possibility of mutating the data structure.  Also, consider `useMemo()` on `COLUMN_DEFINITIONS` if moved into the component.

Reply:
I agree about avoiding mutability and made them readonly.

I really don't see a reason to move them into the component function as they are just constants. There are multiple drawbacks of doings so:
   * increased complexity of the function which goes against the Kibana TS style guide - https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#user-content-write-small-functions
   * small but still impact on memory and performance as even memoization is additional memory and computations
   * impacts reusability - I'm pretty sure those translations (especially OS) will be reused for flyout and form
   * even readability suffers a bit I'd say - more indentation (means prettier more likely to wrap things to next lines), distracts from reading what component actually does

I wold like to have a chat about this separately from this PR - I see Kibana style guide does promote good practices on code complexity, while our code is often growing outside the recommended boundaries. We have really large components (EndpointsList is almost 400 lines long which is almost 30 times bigger than suggested by guidelines) and don't employ decomposition into smaller functions and components.